doc/apiv3-documentation.apib

```diff
@@ -3390,41 +3430,306 @@ For more details and all possible responses see the general specification of [Fo
 
     [Example Form][]
 
-## Add Watcher [/api/v3/work_packages/{work_package_id}/watchers]
++ Response 403 (application/hal+json)
 
-## Add watcher [POST]
+    Returned if the work package does not exist or the client does not have sufficient permissions to see it.
+
+    **Required permission:** edit work package
+
+    *Note that you will only receive this error, if you are at least allowed to see the corresponding work package.*
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:MissingPermission",
+                "message": "You are not allowed to edit the specified work package."
+            }
+
++ Response 404 (application/hal+json)
+
+    Returned if the work package does not exist or the client does not have sufficient permissions to see it.
+
+    **Required permission:** view work package
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:NotFound",
+                "message": "The specified work package does not exist."
+            }
+
+## Watchers [/api/v3/work_packages/{work_package_id}/watchers]
+
++ Model
+    + Body
+
+            {
+                "_links": {
+                    "self": { "href": "/api/v3/work_packages/14/watchers" }
+                },
+                "total": 2,
+                "count": 2,
+                "_type": "Collection",
+                "_embedded": {
+                  "elements": [
+                    {
+                        "_type": "User",
+                        "_links": {
+                            "self": {
+                                "href": "/api/v3/users/1",
+                                "title": "John Sheppard - j.sheppard"
+                            },
+                            "lock": {
+                                "href": "/api/v3/users/1/lock",
+                                "title": "Set lock on j.sheppard"
+                                "method": "POST"
+                            },
+                            "delete": {
```

Comment:
I see your point.
I would argue to not address it now, because those link names make perfect sense on a user object alone and are also already in use.

What might make sense in a future revision of the API: Returning watcher objects instead of user objects (e.g. `api/v3/watchers/13`). Database side watchers have a unique id (e.g. if you watch two work packages, there are two watcher objects for your user account) and might make it more clear *what* you are deleting.

However, the initial thought of not exposing watcher objects was to provide an abstraction from the database concept and call everything a user. This is now one caveat.

**tl;dr:** For now I would leave it as-is, but if it turns out that there will be more confusion that gains, we might want to reverse course...